src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparamorder/InputJavadocParamOrderRecords.java

```diff
@@ -0,0 +1,83 @@
+//non-compiled with javac: Compilable with Java15
+package com.puppycrawl.tools.checkstyle.checks.javadoc.javadocparamorder;
+
+import java.util.List;
+import org.w3c.dom.Node;
+
+/**
+ * Config: default
+ * Some Javadoc.
+ *
+ * @since something
+ */
+class InputJavadocParamOrderRecords {
+
+    /**
+     * Test case for record.
+     *
+     * @param name
+     *            Parameter name desc.
+     * @param <address>
+     *            Parameter address desc. // violation
+     */
+    record Person(String name, String address) {
+        /**
+         * Test case for constructor.
+         *
+         * @param name
+         *            Parameter name desc.
+         */
+        Person(String name) {
+            this(name, "Unknown");
+        }
+    }
+
+    /**
+     * Description.
+     *
+     * @param brand
+     *            Parameter brand desc.
+     * @param licensePlate
+     *            Parameter licensePlate desc.
+     */
+    record Vehicle(String brand, String licensePlate) {
+
+        /**
+         * Description.
+         *
+         * @param <brand
```

Comment:
I am not sure if this should be a violation. This `@param` tag does not have the same name as the constructor parameter. I think the scope of this check should be smaller, in that we only validate tags that exist. If not, this introduces a bunch of grey areas. The compiler will enforce the type definition to be correct; we should simply take the type parameters, parameters, or record components from the definition, and then check if the corresponding `@param` tags exist (basically, take the "intersection" the two sets of information), and are in the proper order. We have another check, [JavadocMethod](https://checkstyle.sourceforge.io/config_javadoc.html#JavadocMethod) to verify the presence of `@param` tags.

Reply:
> I am not sure if this should be a violation

yes there is a violation it should be just `brand` not `<brand` in javadoc

Reply:
> yes there is a violation it should be just `brand` not `<brand` in javadoc

This is the violation that you are printing:

 `Wrong order for "<brand"  tag.`

If `<brand` parameter  does not exist, how can it be out of order?

Reply:
check logic didnt know this it will check whether `<brand` is equal to `brand` if its not it will log violation.

Reply:
This is a case of a missing tag, and is not in the scope of this check.

Reply:
i get it now what u r saying but i need to ask do you want me to change the check to make it like u r expecting?

Reply:
Let's just focus on the test inputs right now, and do no further development of the check. Go through all test inputs, and make sure that `violation` and `ok` comments are placed according to the logic that is discussed in the issue, not what the check is currently producing.

Reply:
done

Reply:
what do we do next, should i begin development of check with new scope as discussed in the issue?

Reply:
Please complete missing items so that we can see correct test cases; this way we will have a clear understanding of check behavior before moving forward. You might be missing my comments because you are prematurely "resolving" items that are not resolved, such as https://github.com/checkstyle/checkstyle/pull/9699#discussion_r623421702 and https://github.com/checkstyle/checkstyle/pull/9699#discussion_r623419337

Reply:
Resolved.